skrub/_join_aggregator.py

```diff
@@ -0,0 +1,405 @@
+        table = table.groupby(cols_to_join).agg(all_ops)
+
+        # flattening post-processing of mode() cols
+        flatten_ops = []
+        for col in [*num_mode_cols, *categ_mode_cols]:
+            flatten_ops.append(pl.col(col).list[0].alias(col))
+        table = table.with_columns(flatten_ops)
+
+        cols_renaming = {
+            col: f"{col}{suffix}" for col in table.columns if col not in cols_to_join
+        }
+        table = table.rename(cols_renaming)
+
+        return table
+
+    def join(self, left, right, left_on, right_on):
+        return left.join(
+            right,
+            how="left",
+            left_on=left_on,
+            right_on=right_on,
+        )
+
+
+class JoinAggregator(BaseEstimator, TransformerMixin):
+    """Perform aggregation on auxiliary dataframes before joining
+    on the base dataframe.
+
+    Apply numerical (mean, std, sum, min, max) and categorical (mode)
+    aggregation operations on the columns to aggregate, selected by dtypes.
+
+    The grouping columns used during the aggregation are the columns used
+    as keys for joining.
+
+    Uses :obj:`~pandas.DataFrame` inputs only.
+
+    Parameters
+    ----------
+    tables : list of tuples
+        List of (dataframe, columns_to_join, columns_to_agg) tuple
+        specifying the auxiliary dataframes and their columns for joining
+        and aggregation operations.
+
+        dataframe : :obj:`~pandas.DataFrame` or str
+            The auxiliary data to aggregate and join.
+            'X' can be provided as a string to perform self-aggregation on
+            the input data.
+
+        columns_to_join : str or array-like
+            Select the columns from the dataframe to use as keys during
+            the join operation.
+
+        columns_to_agg : str or array-like
+            Select the columns from the dataframe to use as values during
+            the aggregation operations.
+
+    main_key : str or array-like
+        Select the columns from the base table to use as keys during
+        the join operation.
+        If main_key refer to a single column, it will be used to join all tables.
+        Otherwise, main_key must specify the key for each table to join.
+
+    agg_ops : str or list of str, default=None
+        Aggregation operations to perform on the auxiliary table.
+        Options: {'mean', 'std', 'min', 'max', 'mode'}. If set to None,
+        ['mean', 'mode'] will be used.
+
+    suffixes : list of str, default=None
+        The suffixes that will be add to each table columns in case of
+        duplicate column names, similar to `("_x", "_y")` in :obj:`~pandas.merge`.
+        If set to None, we will use the table index, by looking at their
+        order in 'tables', e.g. for a duplicate columns: price, price_1, price_2.
+
+    See Also
+    --------
+    :class:`FeatureAugmenter` :
+        Augment a main table by automatically joining multiple
+        auxiliary tables on it.
+    """
+
```

Comment:
It would be great to add here a small reproducible example with some generated data, so that people see the basic syntax to use when join-aggregating